Bcast receiver comms

theJenix opened the pull request:
Groundwork for easily sending broadcast intents between components.  Pay particular attention to BroadcastRegistrar, BroadcastIntent, and IBroadcastActionHandler.

theJenix committed:
Added BroadcastIntent and BroadcastRegistrar classes, to make sending and
receiving broadcast intents easier.  Modified AudioPlayerService and other
similar classes to use Broadcast intents intead of callbacks (as an example).

theJenix committed:
Merge branch 'master' of git://github.com/TheLastCrusade/fanClub into bcast_receiver_comms

bryansills commented on fanClub/src/com/lastcrusade/fanclub/components/PlaybarFragment.java:
```diff
@@ -10,50 +10,29 @@
+     * Broadcast action sent when the Audio Player service is asked to skip a song.
+     * 
+     */
+    public static final String ACTION_SKIP  = AudioPlayerService.class.getName() + ".action.Skipping";
```
for this and ACTION_PLAY should it be skipping and playing or skip and play?

bryansills replied:
Also do we need these lines in this file? Couldn't we just reference the strings in the AudioPlayerService class?

theJenix committed:
Modified PlaybarFragment and playlist service code to remove intent actions, and
use service interface.

Rolled AudioPlayerService into PlaylistService (playlist functionality should be
pulled into PlaylistManager)

theJenix committed:
added todos regarding register/unregisterReceivers

reidbaker commented on fanClub/src/com/lastcrusade/fanclub/components/PlaybarFragment.java:
```diff
@@ -2,61 +2,29 @@
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.playlistServiceLocator = new ServiceLocator<PlaylistService>(this.getActivity(), PlaylistService.class, PlaylistServiceBinder.class);
```
Long line

theJenix committed:
Split up super long line

theJenix committed:
Replaced AudioPlayerService with PlaylistService
